integrations/huggingface-transformers/tutorials/text-classification/text-classification-cli.md

```diff
@@ -0,0 +1,583 @@
+from pprint import pprint
+
+emotion = load_dataset("tweet_eval", "emotion")
+print(emotion)
+pprint(emotion["train"][0])
+```
+
+We can see that each row dataset contains a `text` field which is a string representing the sequence to be classified and a `label` field which is in `{0,1,2,3}` representing
+one of four emotions.
+
+### Using a Hugging Face Dataset Identifier
+
+We can easily pass the `tweet_eval` Hugging Face dataset identifier to the script. Simply replace the `--task_name` argument with `--dataset_name tweet_eval --dataset_config_name emotion --input_column_names text --label_column_name label`. Since the TweetEval dataset contains multiple subsets (e.g. there is a subset
+that classifies text into an emoji), we pass the `--dataset_config_name` to specify the subset. SparseML will then download the dataset from the Hugging Face hub and run training as before.
+
+Run the following to kick off the training process:
+```
+sparseml.transformers.text_classification \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --recipe zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none \
+  --recipe_args '{"num_epochs":12,"qat_start_epoch":7.0, "observer_epoch": 11.0}' \
+  --distill_teacher disable \
+  --dataset_name tweet_eval --dataset_config_name emotion \
+  --input_column_names "text" --label_column_name "label" \
+  --output_dir sparse_quantized_bert-text_classification_tweet_eval_emotion \
+  --do_train --do_eval --max_seq_length 128 --evaluation_strategy epoch --logging_steps 1000 --save_steps 1000 \
+  --per_device_train_batch_size 8 --per_device_eval_batch_size 32 --gradient_accumulation_steps 4 --preprocessing_num_workers 32 \
+  --seed 5114
+```
+
+You will notice that we used the same recipe as we did in the SST2 case (identified by the SparseZoo stub `zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none`). Since the TweetEval Emotion dataset is a single sentence multi-class classification problem, we used the transfer learning recipe from the sentiment analysis task (a single sentence binary classification problem) as the starting point.
+
+#### Recipe Args
+
+To update a recipe and experiment with hyperparameters, you can download the YAML file from SparseZoo, make updates to the YAML directly, and pass the local path to SparseML.
+
+Alternatively, you can use `--recipe_args` to modify the recipe on the fly. In this case, we used the following to run for 12 epochs instead of 13:
+
+```bash
+--recipe_args '{"num_epochs":12,"qat_start_epoch":7.0, "observer_epoch": 11.0}'
+```
+
+### Using Local CSV/JSON Files
+
+Let's walk through how to pass a CSV/JSON dataset to the CLI.
+
+#### Save Dataset as a CSV File
+
+For this example, we use Hugging Face `datasets` to create a CSV file for TweetEval  that can be passed to SparseML's CLI but you can use any framework you want to create the CSV.
```

Comment:
For this example, we use Hugging Face datasets to create a CSV file for TweetEval that can be passed to SparseML's CLI, but you can use any framework you want to create the CSV.